gef.py

```diff
@@ -5237,186 +5421,43 @@ class PCustomCommand(GenericCommand):
 
     def __init__(self) -> None:
         super().__init__(prefix=True)
-        self["struct_path"] = (os.sep.join( (gef.config["gef.tempdir"], "structs")), "Path to store/load the structure ctypes files")
+        self["struct_path"] = (str(pathlib.Path(gef.config["gef.tempdir"]) / "structs"), "Path to store/load the structure ctypes files")
         self["max_depth"] = (4, "Maximum level of recursion supported")
         self["structure_name"] = ("bold blue", "Color of the structure name")
         self["structure_type"] = ("bold red", "Color of the attribute type")
         self["structure_size"] = ("green", "Color of the attribute size")
         return
 
-
-    def do_invoke(self, argv: List[str]) -> None:
-        argc = len(argv)
-        if argc == 0:
+    @parse_arguments({"type": "", "address": ""}, {})
+    def do_invoke(self, *args: Any, **kwargs: Dict[str, Any]) -> None:
+        args = kwargs["arguments"]
+        if not args.type:
             gdb.execute("pcustom list")
             return
 
-        modname, structname = self.get_modulename_structname_from_arg(argv[0])
+        _, structname = self.explode_type(args.type)
```

Comment:
Currently we don't provide any error message when the struct does not exist. It will be helpful to have one that tells the user how to define a struct.

```suggestion
        _, structname = self.explode_type(args.type)

        manager = ExternalStructureManager()
        struct_info = manager.find(structname)
        if not struct_info:
            path = pathlib.Path(gef.config["pcustom.struct_path"]).expanduser()
            struct_color = gef.config["pcustom.structure_type"]
            filename_color = gef.config["pcustom.structure_name"]
            err(f"No struct {Color.colorify(structname, struct_color)} found. "
                f"Use {Color.yellowify('pcustom edit')} or create a file in {Color.colorify(path, filename_color)} "
                 "to define its fields.")
            return
```

Reply:
Actually no, this should be done in `ExternalStructureManger.find` so that it applies for the other `pcustom` subcommands as well. Ignore this suggestion.

Reply:
I think you were right the first time: the caller should handle the error (a-la `str.find()` )

Reply:
Fixed as well, it is the caller's job to check the return value for `ESM.find($structure_name)` so now we have:

```
gef➤  pcustom
[+] Listing custom structures from '/tmp/gef/structs'
 →  /tmp/gef/structs/goo.py (foo_t, goo_t)
gef➤  pcustom meh
[!] No structure named 'goo' found
```